pex/pip/tool.py

```diff
@@ -393,6 +401,36 @@ def _spawn_pip_isolated(
             # `~/.config/pip/pip.conf`.
             pip_args.append("--isolated")
 
+        # Configure a keychain provider if so configured and the version of Pip supports the option.
+        # Warn the user if Pex cannot pass the `--keyring-provider` option and suggest a solution.
+        if package_index_configuration and package_index_configuration.keyring_provider:
+            if self.version.version >= PipVersion.v23_1.version:
+                pip_args.append("--keyring-provider")
+                pip_args.append(package_index_configuration.keyring_provider)
+            else:
+                logger.warning(
+                    "".join(
+                        [
+                            "The PEX_KEYRING_PROVIDER option is set, but Pip v{} does not support ".format(
```

Comment:
Unless I missed something, `PEX_KEYRING_PROVIDER` is not a thing (nor should it be, since this is a build-time only option and not a runtime knob). You probably are thinking in `PIP_KEYRING_PROVIDER` terms. You should be presenting the option the user actually passes to the Pex CLI.